crates/rattler_index/src/lib.rs

```diff
@@ -433,6 +583,8 @@ pub async fn index_s3(
     session_token: Option<String>,
     target_platform: Option<Platform>,
     repodata_patch: Option<String>,
+    write_zst: bool,
+    write_shards: bool,
     force: bool,
```

Comment:
I kinda agree with the clippy error here. This function is growing out of proportions. Perhaps we can introduce a struct here?